lib/SILOptimizer/SILCombiner/SILCombinerApplyVisitors.cpp

```diff
@@ -687,10 +687,18 @@ SILCombiner::buildConcreteOpenedExistentialInfoFromSoleConformingType(
   } else {
     auto ArgType = ArgOperand.get()->getType();
     auto SwiftArgType = ArgType.getASTType();
-    if (!ArgType.isExistentialType() || ArgType.isAnyObject() ||
-        SwiftArgType->isAny())
-      return None;
-    PD = dyn_cast<ProtocolDecl>(SwiftArgType->getAnyNominal());
+    /// If the argtype is an opened existential conforming to a protocol type
+    /// and that the protocol type has a sole conformance, then we can propagate
+    /// concrete type for it as well.
+    ArchetypeType *archetypeTy;
+    if (SwiftArgType->isOpenedExistential() &&
```

Comment:
Added this check for cases where the opened archetype is passed as is to the callee and that the archetype is a protocol type conforming to a sole type.